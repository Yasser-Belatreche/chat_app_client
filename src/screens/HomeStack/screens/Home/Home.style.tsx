import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#3b4ba5',
  },

  happyCoding: {
    color: '#fff',
    fontSize: 30,
    fontWeight: '700',
    marginBottom: 10,
  },

  credit: {
    color: '#fff',
    fontSize: 18,
    fontWeight: '700',
  },

  underline: {
    textDecorationLine: 'underline',
  },
});

export {styles};
