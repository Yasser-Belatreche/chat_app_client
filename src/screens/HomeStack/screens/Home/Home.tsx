import React from 'react';
import {View, Text, Linking} from 'react-native';

// types
import {HomeStackScreenProps} from '../../HomeStack.types';

// styles
import {styles} from './Home.style';

interface Props extends HomeStackScreenProps<'Home'> {}

const Home: React.FC<Props> = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.happyCoding}>Happy Coding ✌️✌️</Text>
      <Text style={styles.credit}>
        <Text>Made By </Text>
        <Text
          style={styles.underline}
          onPress={() =>
            Linking.openURL(
              'https://www.linkedin.com/in/yasser-belatreche-6b450620a/',
            )
          }>
          Yasser
        </Text>
      </Text>
    </View>
  );
};

export {Home};
