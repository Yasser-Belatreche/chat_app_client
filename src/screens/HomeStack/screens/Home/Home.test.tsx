import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
// import {fireEvent, render} from '@testing-library/react-native';

import {Home} from './Home';

describe('HomeStack -> Home Screen', () => {
  const props: any = {};

  it('should renders correctly', () => {
    const tree = renderer.create(<Home {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
