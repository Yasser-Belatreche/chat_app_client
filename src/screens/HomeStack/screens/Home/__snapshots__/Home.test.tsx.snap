// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeStack -> Home Screen should renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3b4ba5",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 30,
        "fontWeight": "700",
        "marginBottom": 10,
      }
    }
  >
    Happy Coding ✌️✌️
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "700",
      }
    }
  >
    <Text>
      Made By 
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      Yasser
    </Text>
  </Text>
</View>
`;
